### PR TITLE
Canvas add drone elements

### LINKED ISSUE #43121: Canvas: Add drone elements
Add drone elements used in December 2021 Hackathon project to the canvas panel. If nothing else this gives us the ability to support the drone simulation dashboard in a production environment.

### PR DESCRIPTION
**Pull request by nmarrs**
**What this PR does / why we need it**:

Add several drone elements to canvas panel that were created as part of December 2021 Hackathon project. Also add drone svg to iot icon list.

https://user-images.githubusercontent.com/22381771/146069028-f84457e5-7336-429b-9c09-87a2532a670d.mov

https://user-images.githubusercontent.com/22381771/146422995-15ff9953-f506-42f8-9824-0add2f360c58.mov

**Which issue(s) this PR fixes**:

Fixes #43121

### CODE REVIEW

File: public/app/features/canvas/elements/droneTop.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+  registerOptionsUI: (builder) => {
+    const category = ['Drone Top'];
+    builder
+      .addCustomEditor({
+        category,
+        id: 'yawAngle',
+        path: 'config.yawAngle',
+        name: 'Yaw Angle',
+        editor: TextDimensionEditor,
```
Comment (ryantxu) [inline public/app/features/canvas/elements/droneTop.tsx:120]: why text dimension?  I would expect something that returns a number

File: public/app/features/canvas/elements/droneFront.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import { TextDimensionEditor } from 'app/features/dimensions/editors';
+import { CanvasElementItem, CanvasElementProps } from '../element';
+import { DimensionContext, TextDimensionConfig } from 'app/features/dimensions';
+
+interface DroneFrontData {
+  rollAngle?: number;
+}
+
+interface DroneFrontConfig {
+  rollAngle?: TextDimensionConfig;
```
Comment (ryantxu) [inline public/app/features/canvas/elements/droneFront.tsx:15]: ScalarDimensionConfig?

File: public/app/features/dimensions/utils.ts [hunk 3]
```diff
@@ -42,6 +44,21 @@ export function getScaleDimensionFromData(
   return getScaledDimension(undefined, cfg);
 }
 
+export function getScalarDimensionFromData(
```
Comment (nmarrs) [inline public/app/features/dimensions/utils.ts:47]: This missing function tripped me up a little bit, kept trying to use getScaleDimension which was resulting in `NaN` being returned

**Review by ryantxu — APPROVED** [review]

**Comment by ryantxu** [comment]
This is a super related item that would be nice to get into core:
https://github.com/grafana/grafana-edge-app/blob/main/src/panels/draw/items/WindTurbine.tsx

One issue is that with this style rotation changes, the values reset to zero when the rotation changes -- must be some way to improve that!

**Comment by nmarrs** [comment]
> One issue is that with this style rotation changes, the values reset to zero when the rotation changes -- must be some way to improve that!

I can look into adding the windmill as well as addressing this issue (same issue for rotor spinning, which especially noticeable at lower speeds), animation state can be a bit more tricky to get right but I'm sure there is a way to clean this up :)

**Pull request merged**